keras/engine/data_adapter.py

```diff
@@ -1033,6 +1033,8 @@ def _convert_single_tensor(x):
 
     if isinstance(x, np.ndarray):
       dtype = None
+      if x.ndim == 1:
+        x = np.expand_dims(x, axis=-1)
```

Comment:
The rank expansion here is not correct, in many cases we want to be able to pass scalar data.

What could be some other solutions?